### PR TITLE
Fix Bug where Password Prompt Failure to Launch could be considered Success

### PR DESCRIPTION
**Pull request by nagilson**
We did not consider failure to run apt-get update as a 'terminal failure', aka a failure that should cause the extension to fail the install request. But updating is often the first thing we do. So included in that request is the portion that launches the elevated master process. If that gets cancelled it will recognize it failed to launch but consider that non terminal so it will timeout. This should fix that.
 
This should be reviewed after https://github.com/dotnet/vscode-dotnet-runtime/pull/1641 is reviewed and merged, which is why it includes those changes.

This should resolve the bad part of https://github.com/dotnet/vscode-dotnet-runtime/issues/1657
Also adds a message to resolve https://github.com/dotnet/vscode-dotnet-runtime/issues/1658

### CODE REVIEW

File: vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts [hunk 1]
```diff
@@ -79,6 +80,8 @@ export class WinMacGlobalInstaller extends IGlobalInstaller {
                 if(conflictingVersion === this.installingVersion)
                 {
                     // The install already exists, we can just exit with Ok.
+                    this.acquisitionContext.eventStream.post(new DotnetAcquisitionAlreadyInstalled(`The .NET SDK ${this.installingVersion} is already installed.`,
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts:83]:
```suggestion
                    this.acquisitionContext.eventStream.post(new DotnetAcquisitionAlreadyInstalled(` `,
```

File: vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts [hunk 2]
```diff
@@ -79,6 +80,9 @@ export class WinMacGlobalInstaller extends IGlobalInstaller {
                 if(conflictingVersion === this.installingVersion)
                 {
                     // The install already exists, we can just exit with Ok.
+                    this.acquisitionContext.eventStream.post(new DotnetAcquisitionAlreadyInstalled(getInstallKeyFromContext(this.acquisitionContext.acquisitionContext) ?? '',
```
Comment (JL03-Yue) [inline vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts:83]: This provides very clear information about what to do if the version is installed. Does the user need to know that the version they required is not downloaded again?

**Review by JL03-Yue — APPROVED** [review]
Looks great to me 🙂

File: vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts (hunk 2 above)
Comment (nagilson) [reply vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts:83]: That's a good point -- it is already there, just in the OutputChannelObserver! :)

```ts
                    this.outputChannel.append(`${
                        (event as DotnetAcquisitionAlreadyInstalled).requestingExtensionId
                    } wants to install .NET ${
                        (event as DotnetAcquisitionAlreadyInstalled).installKey
                    } but it already exists. No downloads or changes were made.\n`);
```

**Pull request merged**